In [1]:
import pandas as pd

df = pd.read_csv("usarrests.csv")

In [2]:
display(df.head())

,Unnamed: 0,Murder,Assault,UrbanPop
0,Alabama,13.2,236.0,58
1,Alaska,10.0,263.0,48
2,Arizona,8.1,294.0,80
3,Arkansas,8.8,190.0,50
4,California,9.0,276.0,91


In [3]:
display(df.describe())

,Murder,Assault,UrbanPop
count,50.00000,49.000000,50.00000
mean,7.78800,182.183673,74.20000
std,4.35551,130.877435,73.40828
min,0.80000,45.000000,6.00000
25%,4.07500,109.000000,53.25000
50%,7.25000,159.000000,66.00000
75%,11.25000,249.000000,77.75000
max,17.40000,879.000000,570.00000


In [4]:
display(df[df["Assault"] > 400])
display(df[df["UrbanPop"] > 100])
display(df[df["UrbanPop"] < 10])

,Unnamed: 0,Murder,Assault,UrbanPop
39,South Carolina,14.4,879.0,48


,Unnamed: 0,Murder,Assault,UrbanPop
14,Iowa,2.2,56.0,570


,Unnamed: 0,Murder,Assault,UrbanPop
31,New York,11.1,254.0,6


In [5]:
df[df["Assault"].isna()]


,Unnamed: 0,Murder,Assault,UrbanPop
9,Georgia,17.4,NaN,60


In [6]:
fill_value = df["Assault"].median()
df["Assault"] = df["Assault"].fillna(fill_value)

In [7]:
print("Median used:", fill_value)
df.isna().sum()

Median used: 159.0


Unnamed: 0    0
Murder        0
Assault       0
UrbanPop      0
dtype: int64

Summary:
Missing value for Assualt, used the median to fill.

Outliers in the data:
Iowa Urbanpop is listed as 570- this is impossible. The percentage cannot exceed 100. I think that this should be flagged for verification and the value set to NaN because it is impossible.
South Carolina assault is at 879, which is possible but seems very unlikely especially since their Urbanpop percent is 48. I would leave the data but flag it for verification.
New York, the same logic applies as South Carolina data. The listed Urbanpop of 6% is very low, but is possible. I would leave the data and flag for verification.